File: CS/09_Forms/AssignIconToButtonField.cs
using Spire.Pdf;
using Spire.Pdf.Fields;
using Spire.Pdf.Graphics;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace AssignIconToButtonField
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a PDF document       
            PdfDocument doc = new PdfDocument();
            PdfPageBase page = doc.Pages.Add();

            //Create a Button
            PdfButtonField btn = new PdfButtonField(page, "button1");
            btn.Bounds = new RectangleF(0, 50, 120, 100);
            btn.HighlightMode = PdfHighlightMode.Push;
            btn.LayoutMode = PdfButtonLayoutMode.CaptionOverlayIcon;

            //Set text and icon for Normal appearance
            btn.Text = "Normal Text";
            btn.Icon = PdfImage.FromFile(@"..\..\..\..\..\..\Data\E-iceblueLogo.png");

            //Set text and icon for Click appearance (Can only be set when highlight mode is Push)
            btn.AlternateText = "Alternate Text";
            btn.AlternateIcon = PdfImage.FromFile(@"..\..\..\..\..\..\Data\PdfImage.png");

            //Set text and icon for Rollover appearance (Can only be set when highlight mode is Push)
            btn.RolloverText = "Rollover Text";
            btn.RolloverIcon = PdfImage.FromFile(@"..\..\..\..\..\..\Data\PDFJAVA.png");

            //Set icon layout
            btn.IconLayout.Spaces = new float[] { 0.5f, 0.5f };
            btn.IconLayout.ScaleMode = PdfButtonIconScaleMode.Proportional;
            btn.IconLayout.ScaleReason = PdfButtonIconScaleReason.Always;
            btn.IconLayout.IsFitBounds = false;

            //Add the button to the document
            doc.Form.Fields.Add(btn);

            String result = "AssignIconToButtonField-result.pdf";

            //Save the document
            doc.SaveToFile(result); 
            
            //Launch the Pdf file
            PDFDocumentViewer(result);

        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }                    
   
        }
    }
}

File: VB/09_Forms/AssignIconToButtonField.vb
Imports Spire.Pdf
Imports Spire.Pdf.Fields
Imports Spire.Pdf.Graphics

Namespace AssignIconToButtonField
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PDF document
            Dim doc As New PdfDocument()

            ' Add a new page to the document
            Dim page As PdfPageBase = doc.Pages.Add()

            ' Create a new button field
            Dim btn As New PdfButtonField(page, "button1")

            ' Set the bounds (position and size) of the button
            btn.Bounds = New RectangleF(0, 50, 120, 100)

            ' Set the highlight mode of the button to "Push"
            btn.HighlightMode = PdfHighlightMode.Push

            ' Set the layout mode of the button to display caption overlaying the icon
            btn.LayoutMode = PdfButtonLayoutMode.CaptionOverlayIcon

            ' Set the caption text of the button
            btn.Text = "Normal Text"

            ' Set the icon image of the button using an image file
            btn.Icon = PdfImage.FromFile("..\..\..\..\..\..\Data\E-iceblueLogo.png")

            ' Set the alternate text of the button
            btn.AlternateText = "Alternate Text"

            ' Set the alternate icon image of the button using an image file
            btn.AlternateIcon = PdfImage.FromFile("..\..\..\..\..\..\Data\PdfImage.png")

            ' Set the rollover text of the button
            btn.RolloverText = "Rollover Text"

            ' Set the rollover icon image of the button using an image file
            btn.RolloverIcon = PdfImage.FromFile("..\..\..\..\..\..\Data\PDFJAVA.png")

            ' Set the spaces between the icon and the caption
            btn.IconLayout.Spaces = New Single() {0.5F, 0.5F}

            ' Set the scale mode for the button icon as proportional
            btn.IconLayout.ScaleMode = PdfButtonIconScaleMode.Proportional

            ' Set the reason for scaling the button icon always
            btn.IconLayout.ScaleReason = PdfButtonIconScaleReason.Always

            ' Disable fitting the icon within the button bounds
            btn.IconLayout.IsFitBounds = False

            ' Add the button field to the document's form fields collection
            doc.Form.Fields.Add(btn)

            ' Set the output file path
            Dim result As String = "AssignIconToButtonField-result.pdf"

            ' Save the modified document to the output file
            doc.SaveToFile(result)

            ' Close the document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer(result)

        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try

        End Sub
    End Class
End Namespace
